openpype/hosts/blender/plugins/publish/validate_object_mode.py

```diff
@@ -10,22 +12,18 @@ class ValidateObjectIsInObjectMode(pyblish.api.InstancePlugin):
     order = pyblish.api.ValidatorOrder - 0.01
     hosts = ["blender"]
     families = ["model", "rig", "layout"]
-    category = "geometry"
+    category = "cleanup"
     label = "Validate Object Mode"
-    actions = [openpype.hosts.blender.api.action.SelectInvalidAction]
+    actions = [SelectInvalidAction]
     optional = False
 
-    @classmethod
+    @staticmethod
     def get_invalid(cls, instance) -> List:
         invalid = []
-        for obj in [obj for obj in instance]:
-            try:
-                if obj.type == 'MESH' or obj.type == 'ARMATURE':
-                    # Check if the object is in object mode.
-                    if not obj.mode == 'OBJECT':
-                        invalid.append(obj)
-            except Exception:
-                continue
+        for obj in set(instance):
+            if isinstance(obj, bpy.types.Object):
+                if not obj.mode == 'OBJECT':
+                    invalid.append(obj)
```

Comment:
❤️ Nice cleanup compared to the original code.

I suspect that now we might be checking more objects than before. So technically this might give different results in production?

Reply:
bpy.types.Object type could be `‘MESH’, ‘CURVE’, ‘SURFACE’, ‘META’, ‘FONT’, ‘CURVES’, ‘POINTCLOUD’, ‘VOLUME’, ‘GPENCIL’, ‘ARMATURE’, ‘LATTICE’, ‘EMPTY’, ‘LIGHT’, ‘LIGHT_PROBE’, ‘CAMERA’, ‘SPEAKER’`
All of then could and should be in mode `'OBJECT'` and most of them have only `'OBJECT'` mode so results should be good for all cases.